lib/msal-angular/src/msal.interceptor.config.ts

```diff
@@ -3,10 +3,14 @@
  * Licensed under the MIT License.
  */
 
+import { HttpRequest } from "@angular/common/http";
 import { PopupRequest, RedirectRequest, InteractionType, SilentRequest } from "@azure/msal-browser";
+import { MsalService } from "./msal.service";
+
+export declare type MsalInterceptorAuthRequest = Omit<PopupRequest, "scopes"> | Omit<RedirectRequest, "scopes"> | Omit<SilentRequest, "scopes"> ;
 
 export type MsalInterceptorConfiguration = {
     interactionType: InteractionType.Popup | InteractionType.Redirect;
     protectedResourceMap: Map<string, Array<string>>;
-    authRequest?: Omit<PopupRequest, "scopes"> | Omit<RedirectRequest, "scopes"> | Omit<SilentRequest, "scopes">;
+    authRequest?: MsalInterceptorAuthRequest | ((originalAuthRequest: MsalInterceptorAuthRequest, msalService: MsalService, req: HttpRequest<unknown>) => MsalInterceptorAuthRequest);
```

Comment:
I guess I can see why it would be useful, but I would prefer it is put at the end:

```suggestion
    authRequest?: MsalInterceptorAuthRequest | ((msalService: MsalService, req: HttpRequest<unknown>, originalAuthRequest: MsalInterceptorAuthRequest) => MsalInterceptorAuthRequest);
```

Reply:
My original comment got lost somewhere, but this makes it possible to use the account selection logic from the interceptor and just override the properties you want.

If we remove the original auth request the developer is required to do the account selection, and this becomes more an provide your own implementation instead of change some properties to your liking.

By providing both the MSAL service and the computed authentication request the developer has a choice to either set own implementation or just override for instance the authority based on the selected account

Reply:
I’ll move it to the back and update the code calling this, tomorrow from behind my Pc cannot do that on my tablet while being on the sofa.